lib/propolis/src/enlightenment/hyperv/mod.rs

```diff
@@ -358,16 +495,57 @@ impl MigrateMulti for HyperV {
             }
         }
 
+        // This enlightenment assumes that if a VM is exported and imported, the
+        // caller asking to import will configure the bhyve VM so that it has
+        // the same apparent guest frequency and offset the VM had when it was
+        // exported. Thus it's not necessary to change the reference TSC page's
+        // contents here. See the TSC module documentation for more details.
+        let msr_reference_tsc = MsrReferenceTscValue(data.msr_reference_tsc);
+        let tsc_overlay = overlays
+            .iter()
+            .position(|page| page.kind() == OverlayKind::ReferenceTsc)
+            .map(|pos| overlays.swap_remove(pos));
+
+        if msr_reference_tsc.enabled() {
+            let Some(overlay) = &tsc_overlay else {
```

Comment:
This is not actually an error if the GPA in the MSR is out of range. (This is different from the hypercall MSR case, where register values with an invalid GPA raise #GP instead of writing back to the MSR.) This might call for a mechanism that simply checks if an overlay *could be created* at a given GPA but that doesn't actually create one.

Reply:
Fixed in 908237b